.azure/pipelines/jobs/default-build.yml

```diff
@@ -41,217 +41,223 @@
-    - ${{ if eq(parameters.agentOs, 'Windows') }}:
-      - task: NuGetCommand@2
-        displayName: 'Clear NuGet caches'
-        condition: succeeded()
-        inputs:
-          command: custom
-          arguments: 'locals all -clear'
-    - ${{ if and(eq(parameters.installJdk, 'true'), eq(parameters.agentOs, 'Windows')) }}:
-      - powershell: ./eng/scripts/InstallJdk.ps1
-        displayName: Install JDK 11
-      - ${{ if eq(parameters.isTestingJob, true) }}:
-        - powershell: |
-            Write-Host "##vso[task.setvariable variable=SeleniumProcessTrackingFolder]$(BuildDirectory)\artifacts\tmp\selenium\"
-            ./eng/scripts/InstallGoogleChrome.ps1
-          displayName: Install Chrome
-    - ${{ if and(eq(parameters.installTar, 'true'), eq(parameters.agentOs, 'Windows')) }}:
-      - powershell: ./eng/scripts/InstallTar.ps1
-        displayName: Find or install Tar
-
-    - ${{ parameters.beforeBuild }}
-
-    - ${{ if ne(parameters.steps, '')}}:
-      - ${{ parameters.steps }}
-    - ${{ if eq(parameters.steps, '')}}:
-      - ${{ if ne(variables['System.TeamProject'], 'public') }}:
-        - ${{ if eq(parameters.agentOs, 'Windows') }}:
-          - ${{ if ne(variables['System.TeamProject'], 'public') }}:
-            - task: PowerShell@2
-              displayName: Setup Private Feeds Credentials
+            queue: BuildPool.Server.Amd64.VS2019
+      variables:
+        - AgentOsName: ${{ parameters.agentOs }}
+        - ASPNETCORE_TEST_LOG_MAXPATH: "200" # Keep test log file name length low enough for artifact zipping
+        - BuildScript: ${{ parameters.buildScript }}
+        - BuildScriptArgs: ${{ parameters.buildArgs }}
+        - _BuildConfig: ${{ parameters.configuration }}
+        - BuildConfiguration: ${{ parameters.configuration }}
+        - BuildDirectory: ${{ parameters.buildDirectory }}
+        - DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+        - TeamName: AspNetCore
+        - ? ${{ if and(eq(parameters.installJdk, 'true'), eq(parameters.agentOs, 'Windows')) }}
+          : - JAVA_HOME: $(Agent.BuildDirectory)\.tools\jdk\win-x64
+        - ? ${{ if or(ne(parameters.codeSign, true), ne(variables['System.TeamProject'], 'internal')) }}
+          : - _SignType: ""
+        - ? ${{ if and(eq(parameters.codeSign, true), eq(variables['System.TeamProject'], 'internal')) }}
+          : - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
+                - _SignType: real
+            - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+                - _SignType: test
+      steps:
+        - checkout: self
+          clean: true
+        - ? ${{ if and(eq(parameters.agentOs, 'Windows'), eq(parameters.isTestingJob, true)) }}
+          : - powershell: ./eng/scripts/InstallProcDump.ps1
+              displayName: Install ProcDump
+            - powershell: ./eng/scripts/StartDumpCollectionForHangingBuilds.ps1 $(ProcDumpPath)procdump.exe artifacts/dumps/ (Get-Date).AddMinutes(160) dotnet
```

Comment:
would like to learn more about this, such as what happens when the build ends (does it leave stuff running?) does it change machine state in any way that survives a reboot?  does it install anything? , etc.

Reply:
cc @javiercn

Reply:
Also only showing in this PR because of the indentation change. Probably something we should have a separate discussion about.